lib/globus_client/identity.rb

```diff
@@ -7,22 +7,27 @@ def initialize(config)
       @config = config
     end
 
-    def get_identity_id(user_id)
+    # @param user_id [String] the username in the form of an email addresss
+    # @return [Hash] id and status of Globus identity
+    def get_identity(user_id)
       @email = user_id
-
       response = lookup_identity
       UnexpectedResponse.call(response) unless response.success?
 
       data = JSON.parse(response.body)
       extract_id(data)
     end
 
-    def exists?(user_id)
-      get_identity_id(user_id)
-      true
-    # if no active user is returned
-    rescue RuntimeError
-      false
+    # @param user_id [String] the username in the form of an email addresss
+    # @return [Boolean] whether the account has a valid status
+    def valid?(user_id)
+      ["used", "private", "unused"].include?(get_identity(user_id)[:status])
```

Comment:
I'm curious what an invalid status might look like.

Reply:
Looking at the tests, `"closed"` looks like one at least.

Reply:
"closed" is currently the only one, according to Globus docs. https://docs.globus.org/api/auth/reference/#identities_api

Reply:
@lwrubel OK, thanks!